Python/import.c

```diff
@@ -386,12 +386,115 @@ import_get_module(PyThreadState *tstate, PyObject *name)
     return m;
 }
 
+static int
+import_ensure_initialized(PyThreadState *tstate, PyObject *mod, PyObject *name)
+{
+    PyInterpreterState *interp = tstate->interp;
+    PyObject *spec;
+    PyObject *value = NULL;
+
+    _Py_IDENTIFIER(__spec__);
+    _Py_IDENTIFIER(_lock_unlock_module);
+
+    /* Optimization: only call _bootstrap._lock_unlock_module() if
+        __spec__._initializing is true.
+        NOTE: because of this, initializing must be set *before*
+        stuffing the new module in sys.modules.
+    */
+    spec = _PyObject_GetAttrId(mod, &PyId___spec__);
+    if (_PyModuleSpec_IsInitializing(spec)) {
+        value = _PyObject_CallMethodIdOneArg(
+            interp->importlib, &PyId__lock_unlock_module, name);
+        if (value == NULL) {
+            Py_DECREF(spec);
+            return -1;
+        }
+        else {
+            return 0;
+        }
+        Py_DECREF(value);
+    }
+    Py_XDECREF(spec);
+    return 0;
+}
+
+
+/* Remove importlib frames from the traceback,
+ * except in Verbose mode. */
+static void
+remove_importlib_frames(PyThreadState *tstate)
```

Comment:
It is hard to follow the changes when such large function is moved. I would move `PyImport_GetModule` instead.